### ISSUE TITLE
Version 1.2.7 Training Error

### ISSUE THREAD

**Issue by patep34** [weecology/DeepForest#445]
Aloha -

First, I've been using DeepForest for almost a year & I'd like to thank y'all for all your hard work on it!

I've been able to make models, etc with DeepForest version 1.2.4 just fine for a while now. But, I tried a new install of the latest version and it crashes pretty badly. Here are the deets:

**Describe the bug**
See below for the screen output. It seems to happen with the "m.trainer.fit(m)" call.

**Environment:**
 - OS: WSL on Windows 11 with Ubuntu 22.04.2 LTS
 - Python version and environment : Miniconda conda environment with Python 3.11.3 (but it happens w/ every Python version I try)
 - DeepForest version 1.2.7 installed via "pip install deepforest"

**Screenshots - all of DeepForest's output including a "print(m.config)" to see the parameter settings**

```python
Reading config file: /home/akepa/miniconda3/envs/deepforest-3113/lib/python3.11/site-packages/deepforest/data/deepforest_config.yml
No validation file provided. Turning off validation loop
GPU available: True (cuda), used: True
TPU available: False, using: 0 TPU cores
IPU available: False, using: 0 IPUs
HPU available: False, using: 0 HPUs


{'workers': 20, 'devices': 'auto', 'accelerator': 'auto', 'batch_size': 7, 'nms_thresh': 0.05, 'score_thresh': 0.3, 'train': {'csv_file': '/mnt/e/20230508-NotClean-NoRuns/DataSets/Train/training_labels_PT_Tree.csv', 'root_dir': '/mnt/e/20230508-NotClean-NoRuns/DataSets/Train', 'lr': 0.001, 'epochs': 5, 'fast_dev_run': False, 'preload_images': False}, 'validation': {'csv_file': '/mnt/e/20230508-NotClean-NoRuns/DataSets/Validate/validation_labels_PT_Tree.csv', 'root_dir': '/mnt/e/20230508-NotClean-NoRuns/DataSets/Validate', 'iou_threshold': 0.4, 'val_accuracy_interval': 1}, 'gpus': '-1'}
GPU available: True (cuda), used: True
TPU available: False, using: 0 TPU cores
IPU available: False, using: 0 IPUs
HPU available: False, using: 0 HPUs
`Trainer(limit_val_batches=1.0)` was configured so 100% of the batches will be used..
Model from DeepForest release https://github.com/weecology/DeepForest/releases/tag/1.0.0 was already downloaded. Loading model from file.
Loading pre-built model: https://github.com/weecology/DeepForest/releases/tag/1.0.0
LOCAL_RANK: 0 - CUDA_VISIBLE_DEVICES: [0]

  | Name  | Type      | Params

0 | model | RetinaNet | 32.1 M

31.9 M    Trainable params
222 K     Non-trainable params
32.1 M    Total params
128.592   Total estimated model params size (MB)
Sanity Checking DataLoader 0:   0%|                                                                                                  | 0/2 [00:00<?, ?it/s]/home/akepa/miniconda3/envs/deepforest-3113/lib/python3.11/site-packages/pytorch_lightning/utilities/data.py:76: UserWarning: Trying to infer the `batch_size` from an ambiguous collection. The batch size we found is 3. To avoid any miscalculations, use `self.log(..., batch_size=batch_size)`.
  warning_cache.warn(
Sanity Checking DataLoader 0: 100%|██████████████████████████████████████████████████████████████████████████████████████████| 2/2 [00:01<00:00,  1.05it/s]/home/akepa/miniconda3/envs/deepforest-3113/lib/python3.11/site-packages/shapely/set_operations.py:133: RuntimeWarning: invalid value encountered in intersection
  return lib.intersection(a, b, **kwargs)
/home/akepa/miniconda3/envs/deepforest-3113/lib/python3.11/site-packages/shapely/set_operations.py:133: RuntimeWarning: invalid value encountered in intersection
  return lib.intersection(a, b, **kwargs)
/home/akepa/miniconda3/envs/deepforest-3113/lib/python3.11/site-packages/shapely/set_operations.py:133: RuntimeWarning: invalid value encountered in intersection
  return lib.intersection(a, b, **kwargs)
/home/akepa/miniconda3/envs/deepforest-3113/lib/python3.11/site-packages/shapely/set_operations.py:133: RuntimeWarning: invalid value encountered in intersection
  return lib.intersection(a, b, **kwargs)
/home/akepa/miniconda3/envs/deepforest-3113/lib/python3.11/site-packages/shapely/set_operations.py:133: RuntimeWarning: invalid value encountered in intersection
  return lib.intersection(a, b, **kwargs)
/home/akepa/miniconda3/envs/deepforest-3113/lib/python3.11/site-packages/shapely/set_operations.py:133: RuntimeWarning: invalid value encountered in intersection
  return lib.intersection(a, b, **kwargs)
/home/akepa/miniconda3/envs/deepforest-3113/lib/python3.11/site-packages/shapely/set_operations.py:133: RuntimeWarning: invalid value encountered in intersection
  return lib.intersection(a, b, **kwargs)
/home/akepa/miniconda3/envs/deepforest-3113/lib/python3.11/site-packages/shapely/set_operations.py:133: RuntimeWarning: invalid value encountered in intersection
  return lib.intersection(a, b, **kwargs)
/home/akepa/miniconda3/envs/deepforest-3113/lib/python3.11/site-packages/shapely/set_operations.py:133: RuntimeWarning: invalid value encountered in intersection
  return lib.intersection(a, b, **kwargs)
/home/akepa/miniconda3/envs/deepforest-3113/lib/python3.11/site-packages/shapely/set_operations.py:133: RuntimeWarning: invalid value encountered in intersection
  return lib.intersection(a, b, **kwargs)
/home/akepa/miniconda3/envs/deepforest-3113/lib/python3.11/site-packages/shapely/set_operations.py:133: RuntimeWarning: invalid value encountered in intersection
  return lib.intersection(a, b, **kwargs)
/home/akepa/miniconda3/envs/deepforest-3113/lib/python3.11/site-packages/shapely/set_operations.py:133: RuntimeWarning: invalid value encountered in intersection
  return lib.intersection(a, b, **kwargs)
/home/akepa/miniconda3/envs/deepforest-3113/lib/python3.11/site-packages/shapely/set_operations.py:133: RuntimeWarning: invalid value encountered in intersection
  return lib.intersection(a, b, **kwargs)
Traceback (most recent call last):
  File "/mnt/c/SDAV/deepforest_detection_pipeline/train_pytorch_df.py", line 179, in <module>
    m.trainer.fit(m)
  File "/home/akepa/miniconda3/envs/deepforest-3113/lib/python3.11/site-packages/pytorch_lightning/trainer/trainer.py", line 520, in fit
    call._call_and_handle_interrupt(
  File "/home/akepa/miniconda3/envs/deepforest-3113/lib/python3.11/site-packages/pytorch_lightning/trainer/call.py", line 44, in _call_and_handle_interrupt
    return trainer_fn(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/akepa/miniconda3/envs/deepforest-3113/lib/python3.11/site-packages/pytorch_lightning/trainer/trainer.py", line 559, in _fit_impl
    self._run(model, ckpt_path=ckpt_path)
  File "/home/akepa/miniconda3/envs/deepforest-3113/lib/python3.11/site-packages/pytorch_lightning/trainer/trainer.py", line 935, in _run
    results = self._run_stage()
              ^^^^^^^^^^^^^^^^^
  File "/home/akepa/miniconda3/envs/deepforest-3113/lib/python3.11/site-packages/pytorch_lightning/trainer/trainer.py", line 976, in _run_stage
    self._run_sanity_check()
  File "/home/akepa/miniconda3/envs/deepforest-3113/lib/python3.11/site-packages/pytorch_lightning/trainer/trainer.py", line 1005, in _run_sanity_check
    val_loop.run()
  File "/home/akepa/miniconda3/envs/deepforest-3113/lib/python3.11/site-packages/pytorch_lightning/loops/utilities.py", line 177, in _decorator
    return loop_run(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/akepa/miniconda3/envs/deepforest-3113/lib/python3.11/site-packages/pytorch_lightning/loops/evaluation_loop.py", line 122, in run
    return self.on_run_end()
           ^^^^^^^^^^^^^^^^^
  File "/home/akepa/miniconda3/envs/deepforest-3113/lib/python3.11/site-packages/pytorch_lightning/loops/evaluation_loop.py", line 244, in on_run_end
    self._on_evaluation_epoch_end()
  File "/home/akepa/miniconda3/envs/deepforest-3113/lib/python3.11/site-packages/pytorch_lightning/loops/evaluation_loop.py", line 325, in _on_evaluation_epoch_end
    call._call_callback_hooks(trainer, hook_name)
  File "/home/akepa/miniconda3/envs/deepforest-3113/lib/python3.11/site-packages/pytorch_lightning/trainer/call.py", line 190, in _call_callback_hooks
    fn(trainer, trainer.lightning_module, *args, **kwargs)
  File "/home/akepa/miniconda3/envs/deepforest-3113/lib/python3.11/site-packages/deepforest/callbacks.py", line 104, in on_validation_epoch_end
    if isinstance(a, pd.DataFrame):
                  ^
```

Let me know if you need anything else to help diagnose this one. Hope it's an easy fix!
-Pat

**Comment by ethanwhite** [comment]
Also, can you post the details of installed packages on your system using either `pip list` or `conda list` from the command line? Given the changes between 1.2.4 and 1.2.7 one possibility is that we need to update a minimum package version for something and so you have an older version installed than some of the recent updates require.

**Comment by patep34** [comment]
Thanks for providing a little test script for training! That helped me eventually figure out that that my Weights & Biases logger was the cause of the crash. I tried a few different versions of it with the 1.2.7 version of DeepForest and none of them worked - but wandb version 0.13.9 works with deepforest version 1.2.4.

But, even though training works (without using wandb), for a 15 epoch model, the code still spits out this error 48 times during the evaluation phase:

/home/akepa/miniconda3/envs/deepforest-3113/lib/python3.11/site-packages/deepforest/visualize.py:123: UserWarning: No color was provided and the label column is not numeric. Using a single default color.
  warnings.warn("No color was provided and the label column is not numeric. "

After that, it spits out this error 73 times:

/home/akepa/miniconda3/envs/deepforest-3113/lib/python3.11/site-packages/shapely/set_operations.py:133: RuntimeWarning: invalid value encountered in intersection
  return lib.intersection(a, b, **kwargs)

This error was also in my original results where wandb didn't work.

Code snippet of my evaluation call:

```
results = model.evaluate(
    csv_file=validation_annotations_csv,
    root_dir=valid_root_directory,
    iou_threshold=iou_threshold,
    savedir=save_dir,
)
```

Both of these errors also show up at the same place in the Colab but, as with my code, they don't seem to interfere with the overall results.

Thanks for your help - at least I can run the latest version now!

**Comment by patep34** [comment]
1. On the warnings/clutter: I only get those when model.evaluate is called. So, no, they don't show up when using the example above.

2. I managed to get wandb working on my own! I deleted the two "model.config["validation"]" settings I had in my code right below the ["train"] ones and it worked. As long as the necessary variables for model.evaluate are set before it's called, it doesn't crash. I guess "model.config["validation"]" settings don't need to specified for the config anymore?

Anyways - thank you so much for your help!
